src/init.cpp

```diff
@@ -1169,8 +1169,9 @@ bool AppInitParameterInteraction()
     }
 
     fRequireStandard = !gArgs.GetBoolArg("-acceptnonstdtxn", !chainparams.RequireStandard());
-    if (chainparams.RequireStandard() && !fRequireStandard)
+    if (chainparams.NetworkID() == ChainType::MAIN && !fRequireStandard) {
```

Comment:
I generally dislike having conditions in the code that check for a specific chain. What is wrong with leaving this as `chainparams.RequireStandard()`? AFAICT it does the same thing.

Reply:
No it does not do the same thing. This pull request changes regtest such that txs are required to be standard by default. If I left the check as is, it would be impossible to pass in a different setting on the command line.

Reply:
Perhaps split it into a `RequireStandard()` and a `DefaultRequireStandard` chainparams flag then?